src/org/jetbrains/plugins/scala/codeInspection/collections/EmulateFlattenInspection.scala

```diff
@@ -0,0 +1,55 @@
+package org.jetbrains.plugins.scala.codeInspection.collections
+
+import org.jetbrains.plugins.scala.codeInspection.InspectionBundle
+import org.jetbrains.plugins.scala.extensions.ResolvesTo
+import org.jetbrains.plugins.scala.lang.psi.api.expr._
+import org.jetbrains.plugins.scala.lang.psi.api.statements.ScFunction
+
+/**
+  * @author Lukasz Piepiora
+  */
+class EmulateFlattenInspection extends OperationOnCollectionInspection {
+  override def possibleSimplificationTypes: Array[SimplificationType] =
+    Array(FlattenSimplification)
+}
+
+object FlattenSimplification extends SimplificationType {
+
+  override def hint: String = InspectionBundle.message("replace.with.flatten")
+
+  override def getSimplification(expr: ScExpression): Option[Simplification] =
+    expr match {
+      case seqOfSeqs `.flatMap` (identityOperation()) =>
+        Some(replace(expr).withText(invocationText(seqOfSeqs, "flatten")))
+      case _ => None
+    }
+
+  private object identityOperation {
+    def unapply(expr: ScExpression): Boolean = stripped(expr) match {
+      case identity(underscore()) => true
+      case identity() => true
+      case ScFunctionExpr(Seq(x), Some(ResolvesTo(param))) if x == param => true
+      case ScFunctionExpr(Seq(x), Some(identity(ResolvesTo(param)))) if x == param => true
+      case _ => false
+    }
+  }
+
+  private object identity {
+
```

Comment:
Method `identity` may also be used with type argument or in the form of postfix expression. You may create an invocation template for `identity` method from `scala.Predef`, like examples in the package object show. It will handle all such cases. It's `unapplySeq` method match qualifier as a first subpattern, you'll just need to ignore it.